File: code/src/main/java/com/ng/code/util/tree/CodeNode.java
package com.ng.code.util.tree;

import androidx.annotation.Nullable;

import com.chad.library.adapter.base.entity.node.BaseNode;

import java.util.List;

public class CodeNode extends BaseNode {
    public String title;
    public int id;
    public String contentPath;
    public String content;
    public int state;

    @Nullable
    public String getMenu() {
        return contentPath == null ? null : contentPath.split("/")[0];
    }

    public CodeNode(String title, int id, String contentPath, String content) {
        this.title = title;
        this.id = id;
        this.contentPath = contentPath;
        this.content = content;
    }

    public CodeNode(String s, String content) {
        this.title = s;
        this.content = content;
    }

    public String getTitle() {
        return title;
    }


    @Nullable
    @Override
    public List<BaseNode> getChildNode() {
        return null;
    }
}

File: app/src/main/java/com/ng/ngleetcode/ui/page/code/bean/CodeNode.kt
package com.ng.ngleetcode.ui.page.code.bean

import com.chad.library.adapter.base.entity.node.BaseNode

data class CodeNode(
    var title: String = "", var id: Int = 0, var contentPath: String = "", var content: String = ""
) : BaseNode() {

    @JvmField
    var state: Int = 0

    fun getMenuStr(): String? {
        return if (contentPath.isEmpty()) null else contentPath.split("/".toRegex())
            .dropLastWhile { it.isEmpty() }
            .toTypedArray()[0]
    }

    override fun toString(): String {
        return "CodeNode(title='$title', id=$id, contentPath='$contentPath', state=$state, childNode=$childNode)"
    }


    override val childNode: MutableList<BaseNode>? = null


}